gammapy/irf/tests/test_effective_area.py

```diff
@@ -87,7 +87,7 @@ def test_to_table():
 @requires_data()
 def test_to_table_is_pointlike(aeff):
     hdu = aeff.to_table_hdu()
-    assert(bool(hdu.meta.pop("is_pointlike", None)) == False)
+    assert "is_pointlike" in hdu.header
```

Comment:
But it should not be in there, right?

Reply:
You're totally right. I've forgotten the `not`. However, I don't understand why the test passed with this syntax on my machine while it shouldn't have...

Reply:
You mean the `bool(header.pop(key, None)) == False)` version? 

if key is not in header, `None` is returned and `bool(None) is False`

Reply:
No, that one passes as it's correct. I was meaning the `assert "is_pointlike" in hdu.header`, which shouldn't pass at all...

Reply:
> No, that one passes as it's correct.

No it's not correct. It would also pass if `is_pointlike` is in the header but with a `falsy` value, e.g. `False` or an empty string.
The only version where this assertion would raise is if `is_pointlike` is in the header, but with a `True` value.

Reply:
Are you sure you ran the test? It has a `@requires_data` decorator, so it might be skipped.

Reply:
> > No, that one passes as it's correct.
> 
> No it's not correct. It would also pass if `is_pointlike` is in the header but with a `falsy` value, e.g. `False` or an empty string. The only version where this assertion would raise is if `is_pointlike` is in the header, but with a `True` value.

Ah you're right. I wrongly considered only the condition of is_pointlike` = True!

Reply:
> Are you sure you ran the test? It has a `@requires_data` decorator, so it might be skipped.

Aahaa that's why! I forgot to define `GAMMAPY_DATA` and the test was silently skypped.

Reply:
Best to just remove the relience on test data and just create a dummy effective area like this:

```

energy_axis = MapAxis.from_energy_bounds('1 TeV', '10 TeV', nbin=3, name='energy_true')
offset_axis = MapAxis.from_bounds(0 * u.deg, 2 * u.deg, nbin=2, name='offset')

aeff = EffectiveAreaTable2D(data=np.ones((3, 2)) * u.m**2, axes=[energy_axis, offset_axis])
```

Reply:
Thanks @maxnoe for the suggestion.